tests/test_array.py

```diff
@@ -776,3 +776,21 @@ async def test_special_complex_fill_values_roundtrip(fill_value: Any, expected:
     assert content is not None
     actual = json.loads(content.to_bytes())
     assert actual["fill_value"] == expected
+
+
+@pytest.mark.parametrize("shape", [(1,), (2, 3), (4, 5, 6)])
+@pytest.mark.parametrize("dtype", ["uint8", "float32"])
+@pytest.mark.parametrize("array_type", ["async", "sync"])
+async def test_nbytes(
+    shape: tuple[int, ...], dtype: str, array_type: Literal["async", "sync"]
+) -> None:
+    """
+    Test that the ``nbytes`` attribute of an Array or AsyncArray correctly reports the capacity of
+    the chunks of that array.
+    """
+    store = MemoryStore()
+    arr = Array.create(store=store, shape=shape, dtype=dtype, fill_value=0)
+    if array_type == "async":
+        assert arr._async_array.nbytes == np.prod(arr.shape) * arr.dtype.itemsize
```

Comment:
Since this is kind of duplicating the logic in the actual function, is it worth hard coding some expected values in the test (instead of calculating them from the array itself)?

Reply:
if we take the definition of `nbytes` to be `size of each element * number of elements`, then I think it's OK to generate the expected values via that calculation  -- we would do that same calculation anyway to generate our expected values. Also, I personally find magic values in tests to be a bit of code smell, because it's not clear where they came from.